src/ui/public/filter_bar/filter_bar_click_handler.js

```diff
@@ -17,32 +25,21 @@ define(function (require) {
         if (event.point.orig) {
           aggConfigResult = event.point.orig.aggConfigResult;
         } else if (event.point.values) {
-          aggConfigResult = findAggConfig(event.point.values);
+          aggConfigResult = findAggConfigResult(event.point.values);
         } else {
           aggConfigResult = event.point.aggConfigResult;
         }
 
-        function findAggConfig(values) {
-          if (_.isArray(values)) { // point series chart
-            var index = _.findIndex(values, 'aggConfigResult');
-            return values[index].aggConfigResult;
-          }
-          return values.aggConfigResult; // pie chart
-        }
-
-        function findLabel(obj) {
-          // TODO: find out if there is always a fieldFormatter
-          var formatter = obj.aggConfig.fieldFormatter();
-          return formatter(obj.key) === event.label;
-        }
-
         if (aggConfigResult) {
           var isLegendLabel = !!event.point.values;
-          var results = _.filter(aggConfigResult.getPath(), { type: 'bucket' });
+          var aggBuckets = _.filter(aggConfigResult.getPath(), { type: 'bucket' });
 
-          if (isLegendLabel) results = _.filter(results, findLabel); // filter results array by legend label
+          // For legend clicks, use the last bucket in the path
+          if (isLegendLabel) {
+            aggBuckets = _.slice(aggBuckets, aggBuckets.length - 1);
```

Comment:
I'm generally extremely wary about extracting values from an array based on the order of the array, but I don't see a better alternative in this case.  It looks like the tests for `AggConfigResult#getPath()` do specifically verify order, so we're probably alright.